src/Sylius/Bundle/SettingsBundle/Manager/SettingsManager.php

```diff
@@ -71,29 +79,22 @@ class SettingsManager implements SettingsManagerInterface
      */
     protected $eventDispatcher;
 
-    /**
-     * @param SchemaRegistryInterface  $schemaRegistry
-     * @param ObjectManager            $parameterManager
-     * @param RepositoryInterface      $parameterRepository
-     * @param FactoryInterface         $parameterFactory
-     * @param Cache                    $cache
-     * @param ValidatorInterface       $validator
-     * @param EventDispatcherInterface $eventDispatcher
-     */
     public function __construct(
-        SchemaRegistryInterface $schemaRegistry,
-        ObjectManager $parameterManager,
-        RepositoryInterface $parameterRepository,
+        ServiceRegistryInterface $schemaRegistry,
+        ServiceRegistryInterface $resolverRegistry,
+        ObjectManager $settingsManager,
+        FactoryInterface $settingsFactory,
         FactoryInterface $parameterFactory,
-        Cache $cache,
+        SettingsResolverInterface $defaultResolver,
```

Comment:
I had it like that first. But I felt like this was a more flexible approach. Since you could more easily swap out the default resolver with something different. Otherwise you have to first `unregister` the `default` resolver from the registry and then register it again. Right now you can just replace the argument on the settings manager, I felt like this was more developer friendly.